### PR TITLE
HAProxy PROXY protocol support for downstream connections

### PR DESCRIPTION
**Pull request by mikroskeem**
Addresses one half of #677 - allowing Geyser to connect to Velocity or BungeeCord with PROXY protocol enabled.

Checklist:
- [x] Get it working
- [x] Add configuration option
- [x] Upstream modified MCProtocolLib parts

### CODE REVIEW

**Comment by Camotoy** [comment]
I mentioned this on Discord but in case you missed it - this probably would be better upstream at MCProtocolLib, unless Steveice rejects it for some reason.

File: connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java:38]: Historically we have only bumped the config option when there are breaking changes in the config.

File: connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java:456]: Unnecessary new whitespace?

File: connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java
Comment (mikroskeem) [reply connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java]: In that case this comment is bit ambiguous. `// Modify this when you introduce breaking changes into the config` or something

Comment (Camotoy) [reply connector/src/main/java/org/geysermc/connector/configuration/GeyserConfiguration.java]: Feel free to include that in your PR.

**Comment by mikroskeem** [comment]
This PR should be now ready for final reviews and merging.

**Review by Camotoy — APPROVED** [review]
This looks good to me, but I will get another review or two before this is merged.

File: connector/pom.xml
Comment (Camotoy) [inline connector/pom.xml:149]: I got an OK from Redned, so just one more thing. `netty-resolver-dns`'s version here is 4.1.43 but I did see you bumped PacketLib's Netty version. Will this be an issue?

Comment (mikroskeem) [reply connector/pom.xml:149]: In ideal, all dependencies should be kept in sync to avoid potential issues (mentioned one on Discord)

**Pull request merged**
